src/operator/rnn-inl.h

```diff
@@ -576,7 +583,7 @@ class RNNOp : public Operator{
                        req[rnn_enum::kData],
                        req[rnn_enum::kParams],
                        req[rnn_enum::kState],
-                       req[rnn_enum::kStateCell],
+                       req_statecell,
```

Comment:
Use conditional operator here to make code more concise:
`param_.mode == rnn_enum::kLstm ? req[rnn_enum::kStateCell] : NULL `

Reply:
Yeah, ternary is an option.  I can update.